code/io/spacemouse.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include "globalincs/vmallocator.h"
+#include "math/vecmat.h"
+#include "io/timer.h"
+
+struct hid_device_;
+typedef hid_device_ hid_device;
+
+namespace io
+{
+	namespace spacemouse {
+		struct SpaceMouseMovement {
+			vec3d translation;
+			angles rotation;
+
+			void handleNonlinearities(std::array<std::tuple<float, float>, 6>& spacemouse_nonlinearity);
+		};
+
+		struct SpaceMouseDefinition {
+			unsigned int vendorID, productID;
+			size_t buttons;
+			enum class Protocol { CONNEXION_3D_OLD, CONNEXION_3D_NEW } protocol;
+		};
+
+		class SpaceMouse {
+			const SpaceMouseDefinition& m_definition;
+			const int m_pollingFrequency;
+
+			hid_device* m_deviceHandle;
+			SpaceMouseMovement m_current;
+			SCP_vector<bool> m_keypresses;
```

Comment:
Given the weird specification of `std::vector<bool>`, are there any gotchas to using it here?

Reply:
No, shouldn't.
Random access is still available, so it doesn't really concern us here, since we never take a pointer to a value.
But it doesn't matter too much anyways, since the buttons aren't yet parsed from the protocol, and are also as of yet never used. I plan to, eventually, but that's a followup PR